### PR TITLE
add mkcert as integration

### PR DESCRIPTION
**Pull request by jochenmanz**
Ive added mkcert as a basic integration and refaktored the hostctl integration, so both "feel" the same when using them. Feedback is appreciated :)

### CODE REVIEW

**Comment by domenkozar** [comment]
I'd like to have the options not tied to the implementation. This allows to switch to a better/different implement in the future.

That's why we decided to name `hosts` and not `hostctl`. 

I'm thinking how to name this option, `hostCertificates`?

File: examples/mkcert/devenv.nix [hunk 1]
```diff
@@ -0,0 +1,20 @@
+{ pkgs, config, ... }:
+
+{
+  mkcert.domains = [
+    "example.com"
```
Comment (shyim) [inline examples/mkcert/devenv.nix:5]: This interface is like easy, but not flexible. What if I want to have one certificate with `*.example.com` and `example.com`. 

Also the referencing directly into the state feels not like a good ux 🙈. Maybe a helper to assemble the path 🤔

Comment (jochenmanz) [reply examples/mkcert/devenv.nix]: Do you have an example on how to add such a helper for the path? 🤔

**Comment by jochenmanz** [comment]
> I'd like to have the options not tied to the implementation. This allows to switch to a better/different implement in the future.
> 
> That's why we decided to name `hosts` and not `hostctl`.
> 
> I'm thinking how to name this option, `hostCertificates`?

Ive renamed the option to certificates, that can be easily switched later on.

**Comment by domenkozar** [comment]
How does it work when you generate `*.example.com`? Does it create a file named `*.example.com`?

File: src/modules/integrations/mkcert.nix [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+in
+{
+  options = {
+    certificates = lib.mkOption {
+      type = lib.types.listOf lib.types.str;
+      default = [ ];
+      description = "List of domains to generate certificates for.";
+      example = [
+        "example.com"
+        "127.0.0.1"
```
Comment (domenkozar) [inline src/modules/integrations/mkcert.nix:15]:
```suggestion
        "*.example.com"
```

**Comment by jochenmanz** [comment]
> How does it work when you generate `*.example.com`? Does it create a file named `*.example.com`?

Wildcards are generated as: _wildcard.example.com-key.pem and _wildcard.example.com.pem

**Comment by schliesser** [comment]
> > How does it work when you generate `*.example.com`? Does it create a file named `*.example.com`?
> 
> Wildcards are generated as: _wildcard.example.com-key.pem and _wildcard.example.com.pem

You can define the filename for the certificate manually if you want:

`mkcert -cert-file certfile.crt -key-file keyfile.key "example.com"`

**Pull request merged**